probePlugin/src/main/scala/org/virtuslab/ideprobe/handlers/IntelliJApi.scala

```diff
@@ -1,26 +1,31 @@
 package org.virtuslab.ideprobe.handlers
 
-import java.nio.file.Path
-import java.nio.file.Paths
-
-import com.intellij.openapi.application.Application
-import com.intellij.openapi.application.ApplicationManager
-import com.intellij.openapi.application.PathManager
+import java.nio.file.{Path, Paths}
+import com.intellij.openapi.application.{Application, ApplicationManager, PathManager}
 import com.intellij.openapi.diagnostic.Logger
-import com.intellij.openapi.util.Key
-import com.intellij.openapi.util.ThrowableComputable
-import com.intellij.openapi.util.UserDataHolder
+import com.intellij.openapi.util.{Key, ThrowableComputable, UserDataHolder}
 import com.intellij.ui.CheckBoxList
 import javax.swing.JList
-import org.virtuslab.ideprobe.Probe
-
+import org.virtuslab.ideprobe.{Config, EndpointsConfig, Probe}
 import scala.annotation.tailrec
-import scala.concurrent.Await
-import scala.concurrent.Future
+import scala.concurrent.{Await, Future}
 import scala.concurrent.duration.Duration
 import scala.reflect.ClassTag
+import pureconfig.generic.auto._
+import org.virtuslab.ideprobe.jsonrpc.PayloadJsonFormat._
 
-object IntelliJApi extends IntelliJApi
+object IntelliJApi extends IntelliJApi {
+  private var config = EndpointsConfig()
```

Comment:
This certainly is not part of intellij api utils, but a probe thing. Also object is here just to allow avoiding inheritance, nothing should be added to it ever.
Just create a new object in handlers like `ProbeConfig` with `initialize(config: String)` and `get(): Config` (we want general api for future and for extensions that might want to hook in to this config. BackgroundTasks should just `val awaitIdleConfig = ProbeConfig.get()[AwaitIdleConfig]("probe.endpoints.awaitIdle")`